main.go

```diff
@@ -407,38 +418,106 @@ func (b *Backend) updateCallStats(t shortTraceMsg) {
 		c.setTotalCalls(b.Stats.TotCalls)
 		c.setTotalCallFailures(b.Stats.TotCallFailures)
 	}
+	globalConnStatsRWMutex.RUnlock()
```

Comment:
Add `break` inside the loop. I don't suppose the same endpoint is present multiple times in `globalConnStats`

Also this is no longer needed as per above...
```suggestion
```